### PR TITLE
[Cleanup] Move metaseq_cli into metaseq/cli

### PR DESCRIPTION
**Pull request by suchenzang**
Per https://github.com/facebookresearch/metaseq/issues/203

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ properly ended on an interrupt.
 ```bash
 srun --ntasks-per-node 1 --gpus-per-node $MODEL_PARALLEL --nodes 1 --cpus-per-task 8 --mem 400gb \
     --quit-on-interrupt --job-name genwork \
-    python3 -m metaseq_cli.interactive_hosted
+    python3 -m cli.interactive_hosted
```
Comment (stephenroller) [inline docs/api.md:36]: shouldn't it be metaseq.cli.interactive_hosted?

Comment (suchenzang) [reply docs/api.md:36]: yeah stuff is still broken - moving back to draft

**Review by stephenroller — APPROVED** [review]
We'll need to announce this on metaseq users, I believe this will require a fresh `pip install -e .`, and a synchronous change with metaseq-internal.

**Comment by suchenzang** [comment]
Yeah already did the internal change here: https://github.com/fairinternal/metaseq-internal/pull/509

**Pull request merged**